pallets/xtransfer/src/fungible_adapter.rs

```diff
@@ -0,0 +1,118 @@
+use crate::traits::*;
+use assets_registry::AccountId32Conversion;
+use sp_std::{convert::Into, marker::PhantomData, result, vec::Vec};
+use xcm::latest::{prelude::*, Error as XcmError, MultiAsset, MultiLocation, Result as XcmResult};
+use xcm_executor::{traits::TransactAsset, Assets};
+
+const LOG_TARGET: &str = "runtime::fungbible-adapter";
+
+pub struct XTransferAdapter<NativeAdapter, AssetsAdapter, Transactor, NativeChecker>(
+	PhantomData<(NativeAdapter, AssetsAdapter, Transactor, NativeChecker)>,
+);
+
+impl<
+		NativeAdapter: TransactAsset,
+		AssetsAdapter: TransactAsset,
+		Transactor: OnWithdrawn + OnDeposited + OnForwarded,
+		NativeChecker: NativeAssetChecker,
+	> TransactAsset for XTransferAdapter<NativeAdapter, AssetsAdapter, Transactor, NativeChecker>
+{
+	fn can_check_in(_origin: &MultiLocation, what: &MultiAsset) -> XcmResult {
```

Comment:
Yes since we haven't support it so far, it's better to reject them. Already remove those methods in fungible adapter, the default impl of Transact trait will throw an error.